File: DataStructureAndAlgorithm/Graph/FloydWarshall/C++/FloydWarshall.cpp
#include "FloydWarshall.h"
#include <iostream>

void floydWarshall(std::vector<std::vector<int>>& graph) {
    int V = graph.size();
    std::vector<std::vector<int>> dist = graph;

    for (int k = 0; k < V; ++k) {
        for (int i = 0; i < V; ++i) {
            for (int j = 0; j < V; ++j) {
                // If vertex k is on the shortest path from i to j, then update the value of dist[i][j]
                if (dist[i][k] != INF && dist[k][j] != INF && dist[i][k] + dist[k][j] < dist[i][j]) {
                    dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
    }

    // Print the shortest distance matrix
    std::cout << "Shortest distances between every pair of vertices:" << std::endl;
    for (int i = 0; i < V; ++i) {
        for (int j = 0; j < V; ++j) {
            if (dist[i][j] == INF) {
                std::cout << "INF\t";
            } else {
                std::cout << dist[i][j] << "\t";
            }
        }
        std::cout << std::endl;
    }
}

File: DataStructureAndAlgorithm/Graph/FloydWarshall/Python/floyd_warshall.py
def floyd_warshall(graph):
    # graph is a V x V matrix where graph[i][j] is the weight of the edge from i to j
    # If there is no edge, graph[i][j] should be float('inf')
    # graph[i][i] should be 0

    V = len(graph)
    dist = list(map(lambda i: list(map(lambda j: j, i)), graph)) # Make a copy

    for k in range(V):
        for i in range(V):
            for j in range(V):
                dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])

    return dist
